daq/faucet_state_collector.py

```diff
@@ -165,24 +165,85 @@ def get_switch_raw(self, switch_name):
     def get_stack_topo(self):
         """Returns formatted topology object"""
         topo_map = {}
-        topo_obj = self.topo_state
         with self.lock:
-            for link in topo_obj.get(TOPOLOGY_GRAPH, {}).get("links", []):
-                link_obj = {}
-                port_map = link.get("port_map")
-                if port_map:
-                    link_obj["switch_a"] = port_map["dp_a"]
-                    link_obj["port_a"] = port_map["port_a"]
-                    link_obj["switch_b"] = port_map["dp_z"]
-                    link_obj["port_b"] = port_map["port_z"]
-                    link_obj["status"] = None
-                topo_map[link["key"]] = link_obj
+            config_obj = self.system_states.get(FAUCET_CONFIG, {}).get(DPS_CFG, {})
+            path_to_root = self.topo_state.get(ROOT_PATH, {})
+            for start_dp, dp_obj in config_obj.items():
+                for start_port, iface_obj in dp_obj.get("interfaces", {}).items():
+                    peer_dp = iface_obj.get("stack", {}).get("dp")
+                    peer_port = str(iface_obj.get("stack", {}).get("port"))
+                    if peer_dp and peer_port:
+                        link_obj = {}
+                        subkey1 = start_dp+":"+start_port
+                        subkey2 = peer_dp+":"+peer_port
+                        key_order = subkey1 < subkey2
```

Comment:
key_order true?  key_order false?  Consider swap_order? or reverse_order? .. something that makes sense as a boolean.